bitmark-references/bitmark-node-software/bitmark-blockain.md

```diff
@@ -0,0 +1,125 @@
+# Blockchain
+
+## Block
+
+Block is a bunch of transactions that have been added to blockchain. A bitmark block contains data as follows:
+
+  size (bytes)    data
+ --------------  ----------------------------------------------------------------
+   2              block version number
+   2              transaction count inside the block
+   8              block number (block height)
+  32              argon2 hash of previous block header
+  32              sha3 hash of all transactions in the block (merkle root)
+   8              timestamp of block generation time, since 1970-01-01T00:00 UTC
+   8              current difficulty in compact mode
+   8              number to make hash meets difficulty
+   ... to end     transactions
+
```

Comment:
What's the byte order of these? We know Bitcoin is a mess of endianness? Is everything in the Bitmark blockchain ordered in the same way?

Reply:
Little Endian for storage since this is the Native format for AMD64 and AARCH64 (Like Raspberry Pi v3).  Same (LE) goes for packed values like the Varint format used as counts in packed transactions.
Only real exception is that some of the logs are output as big endian numbers e.g., block hash and difficulty value for ease of comparison.

Reply:
It'd be great to mention that.